pkg/protocol/engine/eviction/state.go

```diff
@@ -323,6 +324,23 @@ func (s *State) PopulateFromStorage(latestCommitmentSlot iotago.SlotIndex) {
 	}
 }
 
+func (s *State) Reset() {
+	rootBlocksToClear := make([]iotago.BlockID, 0)
+	s.rootBlocks.ForEach(func(slot iotago.SlotIndex, storage *shrinkingmap.ShrinkingMap[iotago.BlockID, iotago.CommitmentID]) {
```

Comment:
After we reset it like this, for some time we're not going to have root blocks for the earliest part of the active root block window available. I think under certain conditions it might be a problem, and cause discrepancies in block solidity. Wdyt?

Reply:
Let's clear one thing up since the naming is weird: the `lastEvictedSlot` is the last committed slot (we evict delayed).

Let's furthermore recap what root blocks are: Root blocks are accepted blocks of a slot that got referenced by at least one accepted block of a later slot. The reason why we track them is because we want to have a compressed perception of "all blocks that are relevant for the solidification of later slots" and they are mainly used to reduce the size of snapshots and to have a perception of the cutoff for our walks (since there is no longer a single root in the tangle once we snapshot).

We can easily prune all slots that are following our latest commitment because those are blocks that we will have to solidify again and they will become root blocks once they got referenced by accepted blocks of a later slot (through normal means).

Since we do not remove the root blocks from the committed slot, we might have dirty entries here that were referenced by blocks that were eventually orphaned but since those were all accepted blocks of the committed slot, it is not a problem to leave them there as in theory it is valid to reference any accepted block of our committed epoch and "in theory we would have to fill the root blocks of our reset epoch to ALL blocks of that epoch since we don't know what chain we are syncing with and what root blocks this chain requires).

That's why we fill the root blocks as part of the warp-sync process to make sure that all the required root blocks are available.

If we are resetting as part of a restart from a snapshot, then we want to sync with the same chain and just maintain the existing root block entries for that slot that we restored from disk (everything above will already be empty).

So I think this is perfectly fine.

Since we feed the root blocks during the warp sync process, all the missing blocks of the new chain are being 

We don't commit to them so having a wrong perception is only bad because of solidification issues. 

If we reset the slots that are larger than our latest commitment, then this is fine because we have not seen any blocks in those slots yet.

The reason why we track them is to make the commitments 
2. if we want to start with say last committed slot 15 then we want to be able to process blocks that reference ANY of the blocks in 15. To not have to store a copy of all blocks, we track which 

We accordingly delete all root blocks of slots above 15. What can happen is that the amount of root blocks that we have in here is larger than it should be because we evicted some accepted blocks in the 


Since root blocks are accepted blocks of a slot that have been referenced by accepted blocks of following slots.

We want to:

1. keep the perception